src/Raven.Server/Commercial/LetsEncrypt/LetsEncryptSimulationHelper.cs

```diff
@@ -174,6 +175,10 @@ public static async Task SimulateRunningServer(ServerStore serverStore, X509Cert
         {
             if (webHost != null)
                 await webHost.StopAsync(TimeSpan.Zero);
+            Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
```

Comment:
lets use a event here and pass a the port to it
serverStore.Server._forTestingPurposes?.ReserverSocketForPortEvent?.Invoke(port)